src/semantics/expression_builder.ts

```diff
@@ -1270,6 +1281,8 @@ export function newCastValue(
                     );
                 }
             }
+            // when the 'meta' of r-value is modified, the 'typeId' of r-value also needs to be modified.
```

Comment:
Another thing is L1268, seems we always push `undefined` to the end of the object literal, but the actual missing field may be in the middle of the interface, e.g.:

``` TypeScript
interface I {
    x?: number
    y?: boolean
    z?: stirng
}

let obj: I = {
    x: 1,
    z: 'h',
}
```

Adding 'z' to the end of the object literal can't ensure its typeid to be equal with interface `I`